test/middleware.test.js

```diff
@@ -0,0 +1,70 @@
+/*global describe, beforeEach, it */
+'use strict';
+var path = require('path');
+var helpers = require('yeoman-generator').test;
+var SANDBOX = path.resolve(__dirname, 'sandbox');
+var fs = require('fs');
+var expect = require('chai').expect;
+var common = require('./common');
+
+describe('loopback:middleware generator', function() {
+  beforeEach(common.resetWorkspace);
+  beforeEach(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
+
+  beforeEach(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
+  });
+
+  it('adds a new phase to server/middleware.json', function(done) {
+    var modelGen = givenMiddlewareGenerator();
+    helpers.mockPrompt(modelGen, {
+      name: 'my-middleware-1',
+      phase: 'my-phase',
+      paths: ['/x', '/y'],
+      params: '{"z": 1}'
+    });
+
+    var builtinSources = Object.keys(readMiddlewaresJsonSync('server'));
+    modelGen.run(function() {
+      var newSources = Object.keys(readMiddlewaresJsonSync('server'));
+      var expectedSources = builtinSources.concat(['my-phase']);
+      expect(newSources).to.have.members(expectedSources);
+      done();
+    });
+  });
+
+  it('adds a new entry to an existing phase in server/middleware.json',
+    function(done) {
+      var modelGen = givenMiddlewareGenerator();
+      helpers.mockPrompt(modelGen, {
+        name: 'my-middleware-2',
+        phase: 'routes',
+        paths: ['/x', '/y'],
+        params: '{"z": 1}'
+      });
+
+      var builtinSources = Object.keys(
+        readMiddlewaresJsonSync('server').routes);
+      modelGen.run(function() {
+        var newSources = Object.keys(readMiddlewaresJsonSync('server').routes);
+        var expectedSources = builtinSources.concat(['my-middleware-2']);
+        expect(newSources).to.have.members(expectedSources);
+        done();
+      });
+    });
+
+  function givenMiddlewareGenerator(dsArgs) {
+    var path = '../../middleware';
+    var name = 'loopback:middleware';
+    var gen = common.createGenerator(name, path, [], dsArgs, {});
+    return gen;
+  }
+
+  function readMiddlewaresJsonSync(facet) {
+    var filepath = path.resolve(SANDBOX, facet || 'server', 'middleware.json');
+    var content = fs.readFileSync(filepath, 'utf-8');
+    return JSON.parse(content);
```

Comment:
You can use `fs.readJsonFileSync(filepath)` here, providing that you change `var fs` to `require('fs-extra')`.